### DISCUSSION TITLE
[indexer] Get ready to nearcore 1.24.0

### DISCUSSION THREAD

**Discussion by khorolets** [near/nearcore#6101] [Node Public Interfaces]
# TL;DR: `d8a593d4f09339ced40aafbd9ecec8b39ea48d41 `

`nearcore` 1.24.0 release has been created.

It is planned to be deployed:

* `testnet` Jan 17, 2022 (**has been deployed, please upgrade your nodes ASAP**)
* `mainnet` Feb 14, 2022

### Quick info
**Commit hash** [`d8a593d4f09339ced40aafbd9ecec8b39ea48d41 `](https://github.com/near/nearcore/commit/d8a593d4f09339ced40aafbd9ecec8b39ea48d41) 1.24.0-rc.4
**Expected issues**: NEAR Indexer Framework contains small changes on `init_configs` function and might require some changes on your side (not relevant if you have updated to 1.24.0-rc.1 before)

```
BEFORE
near_indexer::init_configs(...);

AFTER
near_indexer::init_configs(...).expect("Failed to initiate configs");
```

**Reference: have a look at how [we made it on NEAR Indexer for Explorer](https://github.com/near/near-indexer-for-explorer/pull/247/files)

Feel free to leave your questions in the comments below.

---

<details>
<summary>**Outdated and buggy** Release candidates 1, 2, 3</summary>
## Release candidates 1, 2, 3

> 1.24.0-rc.3 is not a mandatory release. It includes the fix that makes RAM usage more efficient. You can skip this release if you don't care about the RAM usage on 1.24.0-rc.2

### Quick info
**Commit hash** [`09e1708adf503be6015b568462213ba6aa4e7ca1 `](https://github.com/near/nearcore/commit/09e1708adf503be6015b568462213ba6aa4e7ca1) 1.24.0-rc.3
**Expected issues**: NEAR Indexer Framework contains small changes on `init_configs` function and might require some changes on your side (not relevant if you have updated to 1.24.0-rc.1 before)

```
BEFORE
near_indexer::init_configs(...);

AFTER
near_indexer::init_configs(...).expect("Failed to initiate configs");
```

**Reference: have a look at how [we made it on NEAR Indexer for Explorer](https://github.com/near/near-indexer-for-explorer/pull/254/files)

Protocol upgrade is expected, **it means that if you have your indexer outdated it'll be stuck once a protocol will be upgraded**, on `testnet` it should happen Jan 19, 2022, please hurry up.

Also, a database migration will happen it takes about a minute on an archival node.

### Quick info
**Commit hash** [`fdd3cffeab68299a7737ce60ac50a1ecd268f6ad `](https://github.com/near/nearcore/commit/fdd3cffeab68299a7737ce60ac50a1ecd268f6ad) 1.24.0-rc.1
**Expected issues**: NEAR Indexer Framework contains small changes on `init_configs` function and might require some changes on your side

```
BEFORE
near_indexer::init_configs(...);

AFTER
near_indexer::init_configs(...).expect("Failed to initiate configs");
```

**Reference: have a look at how [we made it on NEAR Indexer for Explorer](https://github.com/near/near-indexer-for-explorer/pull/230)
</details>

**Comment by khorolets** [comment]
Post was updated with relevant info about `1.24.0-rc.2`

**Reply by khorolets** [reply]
Post was updated with relevant info about `1.24.0-rc.3`

**Reply by DDeAlmeida** [reply]
Work well ! Thank you

**Comment by khorolets** [comment]
Post is updated with the information about the release been postponed until next Monday

**Comment by khorolets** [comment]
Post was updated with relevant info about `1.24.0-rc.4`

**Comment by khorolets** [comment]
Post was updated with commit hash of the release `1.24.0` - there is no difference with `1.24.0-rc.4`
